src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/cas1/OutOfServiceBedTest.kt

```diff
@@ -1244,6 +1250,164 @@ class OutOfServiceBedTest : InitialiseDatabasePerClassTestBase() {
     }
   }
 
+  @Nested
+  inner class GetOutOfServiceBedWithMultipleRevisionsInAllTemporalities : InitialiseDatabasePerClassTestBase() {
```

Comment:
i appreciate trying to group these tests together, but i think they would be better sitting under the existing GetAllOutOfServiceBeds inner class? I've found that when we have multiple classes testing the same thing I tend to completely miss one of the sets of tests. I'm not sure if you can put nested classes in nested classes, which would be the easiest way to merge them

Does this not duplicate the existing 'Get All Out-Of-Service Beds filters by temporality correctly' test to some degree? Presumably it wasn't straight forward to 'fix' that test to cover the additional scenarios?

Reply:
I reworked the original test code and I think there was an issue in the way it was setting up the test data, with a separate oosb record for each revision instead of 1 oosb record with multiple revisions.  As such, it would always return a response for every temporality and when all 3 temporalities were in the specification it would return 3 separate records rather than limiting the responses to always return only the latest one.